frontend/src/api/actions/init.js

```diff
@@ -32,140 +42,167 @@ function toggleMockApi(enableMockAPI) {
+                    parsedGlobals,
+                    parseURLSearchString(window.location.search)
+                );
+                // Immediately set any global variables in the Redux store.
+                await dispatch(setGlobals(prepareGlobals(parsedGlobals)));
+            }
+        }
 
         if (shouldRunStage("toggleMockAPI")) {
-            toggleMockApi(mockAPI);
+            const globals = globalsSelector(getState());
+            toggleMockApi(globals.mockAPI);
         }
 
         if (shouldRunStage("setActiveUser")) {
             await dispatch(fetchActiveUser());
         }
 
         if (shouldRunStage("setActiveUserRole")) {
-            await dispatch(setActiveUserRole(activeUserRole));
+            const activeRole = activeRoleSelector(getState());
+            await dispatch(setActiveUserRole(activeRole, { skipInit: true }));
         }
 
         if (shouldRunStage("fetchSessions")) {
             await dispatch(fetchSessions());
         }
 
-        if (shouldRunStage("setActiveSession") && activeSession !== null) {
+        if (shouldRunStage("setActiveSession")) {
             // after sessions are fetched, we compare with the active session.
             // The active session might need to be "updated" if the ID matches but
             // the data doesn't
             const state = getState();
             const sessions = sessionsSelector(state);
+            const activeSession = activeSessionSelector(state);
+            // There are two places where the active session could be store:
+            // in the URL as a global and in the Redux store. Prefer the value
+            // in the URL.
             const matchingSession =
-                sessions.filter(s => s.id === activeSession.id)[0] ||
-                activeSession;
+                sessions.find(
+                    session => session.id === parsedGlobals.activeSession
+                ) || activeSession;
 
-            await dispatch(setActiveSession(matchingSession));
+            await dispatch(
+                setActiveSession(matchingSession, { skipInit: true })
+            );
         }
 
         if (shouldRunStage("updateGlobals")) {
-            await dispatch(updateGlobals(mockAPI, activeSession, getState));
+            await dispatch(setGlobals(prepareGlobals(parsedGlobals)));
         }
 
-        const promises = [];
-
-        // `fetchActions` array contains all the fetch API calls that need to be
-        // made in order to obtain all data that the app needs. Each action would
-        // be execute and dispatch later in the `fetchActions.map`.
-        const fetchActions = [
-            fetchApplicants,
-            fetchApplications,
-            fetchAssignments,
-            fetchContractTemplates,
-            fetchInstructors,
-            fetchPositions
-        ];
-
-        fetchActions.map(action => dispatch(action()));
-        await Promise.all(promises);
+        console.log(stage, options);
```

Comment:
Is this `console.log` left here intentionally?

Reply:
Nope. That was a mistake!